test/unit/options/options_black.cpp

```diff
@@ -0,0 +1,168 @@
+                  range.second = *v.maximum;
+                }
+                EXPECT_NO_THROW(d_solver.setOption(
+                    name, std::to_string((range.first + range.second) / 2)));
+              },
+              [this, &name](const OptionInfo::NumberInfo<uint64_t>& v) {
+                std::pair<uint64_t, uint64_t> range{
+                    std::numeric_limits<uint64_t>::min(),
+                    std::numeric_limits<uint64_t>::max()};
+                if (v.minimum)
+                {
+                  EXPECT_THROW(
+                      d_solver.setOption(name, std::to_string(*v.minimum - 1)),
+                      CVC5ApiOptionException);
+                  EXPECT_NO_THROW(
+                      d_solver.setOption(name, std::to_string(*v.minimum)));
+                  range.first = *v.minimum;
+                }
+                if (v.maximum)
+                {
+                  EXPECT_THROW(
+                      d_solver.setOption(name, std::to_string(*v.maximum + 1)),
+                      CVC5ApiOptionException);
+                  EXPECT_NO_THROW(
+                      d_solver.setOption(name, std::to_string(*v.maximum)));
+                  range.second = *v.maximum;
+                }
+                EXPECT_NO_THROW(d_solver.setOption(
+                    name, std::to_string((range.first + range.second) / 2)));
+              },
+              [this, &name](const OptionInfo::NumberInfo<double>& v) {
+                std::pair<double, double> range{
+                    std::numeric_limits<double>::min(),
+                    std::numeric_limits<double>::max()};
+                if (v.minimum)
+                {
+                  EXPECT_THROW(
+                      d_solver.setOption(name, std::to_string(*v.minimum - 1)),
+                      CVC5ApiOptionException);
+                  EXPECT_NO_THROW(
+                      d_solver.setOption(name, std::to_string(*v.minimum)));
+                  range.first = *v.minimum;
+                }
+                if (v.maximum)
+                {
+                  EXPECT_THROW(
+                      d_solver.setOption(name, std::to_string(*v.maximum + 1)),
+                      CVC5ApiOptionException);
+                  EXPECT_NO_THROW(
+                      d_solver.setOption(name, std::to_string(*v.maximum)));
+                  range.second = *v.maximum;
+                }
+                EXPECT_NO_THROW(d_solver.setOption(
+                    name, std::to_string((range.first + range.second) / 2)));
+              },
+              [this, &name](const OptionInfo::ModeInfo& v) {
+                EXPECT_THROW(d_solver.setOption(name, "foobarbaz"),
+                             CVC5ApiOptionException);
+                for (const auto& m : v.modes)
+                {
+                  d_solver.setOption(name, m);
+                }
+              },
+          },
+          info.valueInfo);
+    }
+    catch (const CVC5ApiOptionException&)
```

Comment:
None of these tests should throw and not be wrapped in `ASSERT_THROW` or `EXPECT_THROW`, right?